src/mono/mono/mini/interp/interp.c

```diff
@@ -1692,8 +1731,23 @@ ves_pinvoke_method (
 	mono_arch_set_native_call_context_args (&ccontext, &frame, sig, call_info);
 	args = &ccontext;
 #else
-	InterpMethodArguments *margs = build_args_from_sig (sig, &frame);
-	args = margs;
+
+#ifdef HOST_WASM
+	BuildArgsFromSigInfo *call_info = cache_data->call_info;
+#else
```

Comment:
Isn't this path dead ? Is it supposed to run on WASI ? Shouldn't wasi always be wasm ?

Reply:
Its initialized here:
```
		cache_data->entry_func = (MonoPIFunc)mono_wasm_get_interp_to_native_trampoline (sig);
		cache_data->call_info = get_build_args_from_sig_info (get_default_mem_manager (), sig);
```

On wasm, we need to cache both the entry_func and the call_info, this is why its using a different code path.

Reply:
My point is that this code is in the `!MONO_ARCH_HAVE_INTERP_PINVOKE_TRAMP` path.  This path is wasm only. So my question is why do we guard for `HOST_WASM` again and we also have an else path for it. If anything, I would expect the else path to be a pragma error or something like that.

Reply:
So then the code in the `!HOST_WASM` path here is dead.